Makefile-test.mk

```
@@ -51,6 +51,9 @@ E2E_USE_HELM ?= false
 # Default will delete default kind cluster
 KIND_CLUSTER_NAME ?= kind
 
+# For restricting to a specific directory
+WHAT ?= .
```

```suggestion
GO_TEST_WHAT ?= .
```

Can you restrict this in go test?

---

I wouldn't like to rename this, as this would diverge the interface from what's documented e.g. [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/testing.md#run-unit-tests-from-certain-packages).

Is your concern that `WHAT` is supported only in the `make test` rule, and no other rule in this Makefile? \
If so, please note [this guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/testing.md) also uses `WHAT` only for `make test`.

If I were to rename this, I would consider it healthier not to merge this PR at all.

---

First, our guide is https://github.com/kubernetes-sigs/kueue/blob/main/CONTRIBUTING.md. Your pointed guide is for kubernetes/kubernetes.
Second, we have the target environment vars for integrations and E2E:
- E2E: https://github.com/kubernetes-sigs/kueue/blob/21dbaa606e04d1a57c2fd17a703ff0e3ad48e5ed/Makefile-test.mk#L40
- Integration: https://github.com/kubernetes-sigs/kueue/blob/21dbaa606e04d1a57c2fd17a703ff0e3ad48e5ed/Makefile-test.mk#L32

So, `WHAT` should have a context for Unit Test. It would be better to have `GO_TEST_TARGET` to alisng with https://github.com/kubernetes-sigs/kueue/blob/21dbaa606e04d1a57c2fd17a703ff0e3ad48e5ed/Makefile-test.mk#L19. The `GO_TEST_` prefix indicates for Unit Test.

---

Thank you for the context! I renamed `WHAT` -> `GO_TEST_TARGET`.